src/main/java/core/basesyntax/model/ProductTransaction.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.model;
+
+import core.basesyntax.exception.OperationNotFoundException;
+import java.util.Arrays;
+import java.util.StringJoiner;
+
+public class ProductTransaction {
+    private final Operation operation;
+    private final String product;
+    private final int quantity;
+
+    public ProductTransaction(Operation operation, String product, int quantity) {
+        this.operation = operation;
+        this.product = product;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getProduct() {
+        return product;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public static ProductTransaction of(String operation, String product, String quantity) {
+        Operation operationOperation = Operation.getFromString(operation);
+        int quantityInteger = Integer.parseInt(quantity);
+        return new ProductTransaction(operationOperation, product, quantityInteger);
+    }
+
+    @Override
+    public String toString() {
+        return new StringJoiner(", ", ProductTransaction.class.getSimpleName() + "[", "]")
+                .add("operation='" + operation + "'")
+                .add("product='" + product + "'")
+                .add("quantity=" + quantity)
+                .toString();
+    }
```

Comment:
The same here, if you create report in different class, why do you need toString in your models?